core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/IdentityPlugin.java

```diff
@@ -115,31 +120,32 @@ public void init(SnowOwlConfiguration configuration, Environment env) throws Exc
 			identityProvider = new AdminPartyIdentityProvider(identityProvider);
 		}
 		
-		// ensure that identity providers are initialized
-		identityProvider.init(env);
-		
 		return identityProvider;
 	}
 
 	private boolean isJwksIdentityProviderSet(Collection<IdentityProvider> identityProviders) {
 		return identityProviders.stream().anyMatch(ip -> ip instanceof JwksIdentityProvider);
 	}
 
-	private List<IdentityProvider> createProviders(Environment env, List<IdentityProviderConfig> providerConfigurations) {
+	private List<IdentityProvider> createProviders(ServiceProvider ctx, List<IdentityProviderConfig> providerConfigurations) {
 		final List<IdentityProvider> providers = newArrayListWithExpectedSize(3);
-		env.plugins().getPlugins().stream()
-			.filter(IdentityProviderFactory.class::isInstance)
-			.map(IdentityProviderFactory.class::cast)
-			.forEach(factory -> {
-				Optional<IdentityProviderConfig> providerConfig = providerConfigurations.stream().filter(conf -> conf.getClass() == factory.getConfigType()).findFirst();
-				if (providerConfig.isPresent()) {
-					try {
-						providers.add(factory.create(env, providerConfig.get()));
-					} catch (Exception e) {
-						throw new SnowowlRuntimeException(String.format("Couldn't initialize '%s' identity provider", factory), e);
-					}
+		final List<IdentityProviderFactory> factories = ctx.service(Plugins.class).getPlugins().stream()
+		.filter(IdentityProviderFactory.class::isInstance)
+		.map(IdentityProviderFactory.class::cast)
+		.toList();
+		
+		providerConfigurations.forEach(providerConfig -> {
+			Optional<IdentityProviderFactory> factory = factories.stream().filter(f -> f.getConfigType() == providerConfig.getClass()).findFirst();
+			if (factory.isPresent()) {
+				try {
+					providers.add(factory.get().create(providerConfig));
+				} catch (Exception e) {
+					throw new SnowOwl.InitializationException(String.format("Couldn't initialize '%s' identity provider", factory), e);
 				}
-			});
+			} else {
+				throw new SnowOwl.InitializationException("Couldn't find identity provider factory for '%s'.", providerConfig.getClass());
```

Comment:
It should be a rare occasion that the configuration class can be deserialized correctly, but the corresponding factory can not be found via plugin lookup (in most cases these will be co-located in the same OSGi bundle). This is more like a development-time issue rather than something that would happen in production.

I think it's OK to use the same exception type either way.

Reply:
Yeah, I was thinking of changing this to IllegalStateException to indicate a programmatic related error rather than a configuration or other error.

Reply:
Changed it.